spec/controllers/discovery_controller_spec.rb

```diff
@@ -83,29 +83,36 @@
   end
 
   describe '#keys' do
-    it 'returns the key parameters' do
-      get :keys
-      data = JSON.parse(response.body)
+    subject { get :keys }
 
-      expect(data.sort).to eq({
-        'keys' => [
-          {
-            'kty' => 'RSA',
-            'kid' => 'IqYwZo2cE6hsyhs48cU8QHH4GanKIx0S4Dc99kgTIMA',
-            'e' => 'AQAB',
-            'n' => (
-              'sjdnSA6UWUQQHf6BLIkIEUhMRNBJC1NN_pFt1EJmEiI88GS0ceROO5B5O' \
-              'oo9Y3QOWJ_n-u1uwTHBz0HCTN4wgArWd1TcqB5GQzQRP4eYnWyPfi4Cfe' \
-              'qAHzQp-v4VwbcK0LW4FqtW5D0dtrFtI281FDxLhARzkhU2y7fuYhL8fVw' \
-              '5rUhE8uwvHRZ5CEZyxf7BSHxIvOZAAymhuzNLATt2DGkDInU1BmF75tEt' \
-              'BJAVLzWG_j4LPZh1EpSdfezqaXQlcy9PJi916UzTl0P7Yy-ulOdUsMlB6' \
-              'yo8qKTY1-AbZ5jzneHbGDU_O8QjYvii1WDmJ60t0jXicmOkGrOhruOptw'
-            ),
-            'use' => 'sig',
-            'alg' => 'RS256',
-          }
-        ],
-      }.sort)
+    shared_examples 'a key response' do |options|
+      expected_parameters = options[:expected_parameters]
+
+      it "includes #{expected_parameters.join(', ')} parameters" do
+        subject
+        data = JSON.parse(response.body)
+        key = data['keys'].first
+
+        expected_parameters.each do |param|
+          expect(key[param.to_s]).to be_present
+        end
```

Comment:
Can you change this to make sure *only* the expected parameters are present? i.e. something like:

`expect(key.symbolize_keys.keys).to match_array expected_parameters`